test/extensions/filters/network/common/redis/codec_impl_test.cc

```diff
@@ -427,6 +427,424 @@ TEST_F(RedisEncoderDecoderImplTest, InvalidBulkStringExpectLF) {
+  answer.type(RespType::BulkString);
+  answer.asString() = "42";
+
+  buffer_.add("42\r\n");
+  decoder_.decode(buffer_);
+  EXPECT_EQ(1UL, decoded_values_.size());
+  EXPECT_EQ(RespType::Array, decoded_values_[0]->type());
+  EXPECT_EQ(1UL, decoded_values_[0]->asArray().size());
+  EXPECT_EQ(answer, decoded_values_[0]->asArray()[0]);
+}
+
+TEST_F(RedisEncoderDecoderImplTest, InlineCommandNumericArgument) {
+  RespValue echo;
+  echo.type(RespType::BulkString);
+  echo.asString() = "echo";
+
+  RespValue answer;
+  answer.type(RespType::BulkString);
+  answer.asString() = "42";
+
+  buffer_.add("echo 42\r\n");
+  decoder_.decode(buffer_);
+  EXPECT_EQ(1UL, decoded_values_.size());
+  EXPECT_EQ(RespType::Array, decoded_values_[0]->type());
+  EXPECT_EQ(2UL, decoded_values_[0]->asArray().size());
+  EXPECT_EQ(echo, decoded_values_[0]->asArray()[0]);
+  EXPECT_EQ(answer, decoded_values_[0]->asArray()[1]);
+}
+
+TEST_F(RedisEncoderDecoderImplTest, InlineCommandTrimWhitespace) {
+  RespValue ping;
+  ping.type(RespType::BulkString);
+  ping.asString() = "ping";
+
+  buffer_.add("   ping   \r\n");
+  decoder_.decode(buffer_);
+  EXPECT_EQ(1UL, decoded_values_.size());
+  EXPECT_EQ(RespType::Array, decoded_values_[0]->type());
+  EXPECT_EQ(1UL, decoded_values_[0]->asArray().size());
+  EXPECT_EQ(ping, decoded_values_[0]->asArray()[0]);
+}
+
+TEST_F(RedisEncoderDecoderImplTest, InlineCommandQuotedEmpty) {
+  RespValue ping;
+  ping.type(RespType::BulkString);
+  ping.asString() = "";
+
+  buffer_.add("\"\"\r\n");
+  decoder_.decode(buffer_);
+  EXPECT_EQ(1UL, decoded_values_.size());
+  EXPECT_EQ(RespType::Array, decoded_values_[0]->type());
+  EXPECT_EQ(1UL, decoded_values_[0]->asArray().size());
+  EXPECT_EQ(ping, decoded_values_[0]->asArray()[0]);
+}
+
+TEST_F(RedisEncoderDecoderImplTest, InlineCommandQuotedCommand) {
+  RespValue ping;
+  ping.type(RespType::BulkString);
+  ping.asString() = "ping";
+
+  buffer_.add("\"ping\"\r\n");
+  decoder_.decode(buffer_);
+  EXPECT_EQ(1UL, decoded_values_.size());
+  EXPECT_EQ(RespType::Array, decoded_values_[0]->type());
+  EXPECT_EQ(1UL, decoded_values_[0]->asArray().size());
+  EXPECT_EQ(ping, decoded_values_[0]->asArray()[0]);
+}
+
+TEST_F(RedisEncoderDecoderImplTest, InlineCommandQuotedArgument) {
+  RespValue echo;
+  echo.type(RespType::BulkString);
+  echo.asString() = "echo";
+
+  RespValue hello;
+  hello.type(RespType::BulkString);
+  hello.asString() = "hello world";
+
+  buffer_.add("echo \"hello world\"\r\n");
+  decoder_.decode(buffer_);
+  EXPECT_EQ(1UL, decoded_values_.size());
+  EXPECT_EQ(RespType::Array, decoded_values_[0]->type());
+  EXPECT_EQ(2UL, decoded_values_[0]->asArray().size());
+  EXPECT_EQ(echo, decoded_values_[0]->asArray()[0]);
+  EXPECT_EQ(hello, decoded_values_[0]->asArray()[1]);
+}
+
+TEST_F(RedisEncoderDecoderImplTest, InlineCommandQuotedEmptyArgument) {
+  RespValue echo;
+  echo.type(RespType::BulkString);
+  echo.asString() = "echo";
+
+  RespValue hello;
```

Comment:
nit: may be better to name it "empty_str"

Reply:
Fixed,